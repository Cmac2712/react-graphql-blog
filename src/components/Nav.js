import React, { Component } from 'react';
import Signout from './Signout';
import { Link } from 'react-router-dom';
import User, { CURRENT_USER_QUERY } from './User';

const Nav = props => (
		<nav>
			<User>
				{
					({data, loading}) => {

						if (data && data.me) {
							return (
								<>
								<Link to={`/`}>
									Home
								</Link>
								<Link to={`/createpost`}>
									Create Post
								</Link>
								<Signout/>
							</>
							)
						}

						return (
							<>
								<Link to={`/`}>
									Home
								</Link>
								<Link to={`/account`}>
									Log In / Create an Account
								</Link>
							</>
						)
					}
				}
			</User>
		</nav>
);

export default Nav;
