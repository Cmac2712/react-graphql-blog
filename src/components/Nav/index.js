import React, { Component } from 'react';
import Signout from '../Signout';
import { Link } from 'react-router-dom';
import User, { CURRENT_USER_QUERY } from '../User';
import NavBar from './styles';

const Nav = props => (
	<NavBar
		data-test
	>
			<Link
				className="logo"
				to={`/`}
			>
				React GraphQl Blog
			</Link>
			<User>
				{
					({data, loading}) => {

						if (data && data.me) {
							return (
								<Link to={`/cms/create-post`}>
									Dashboard
								</Link>
							)
						}

						return (
								<Link
									to={`/cms/create-post`}>
									Log In / Create an Account
								</Link>
						)
					}
				}
			</User>
		</NavBar>
);

export default Nav;
