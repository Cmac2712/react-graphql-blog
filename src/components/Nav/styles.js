import styled from 'styled-components';

const NavBar = styled.nav`
	display: flex;
	justify-content: flex-end;
	background-color: ${props => props.theme.brandColor};
	padding: ${props => props.theme.spacingUnit};
	
	a {
		color: #fff;
		padding: 0 ${props => props.theme.spacingUnit};
	}
`;

export default NavBar;
