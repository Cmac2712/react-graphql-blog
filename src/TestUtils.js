import { CURRENT_USER_QUERY } from './components/User';

const mockUser = {
	__typename: "Query", 
	id: "kfjasldkfjaldskfkb",
	email: "fakeUser@fakemail.com", 
	name: "fakeuser123", 
	screenName: "Fake User", 
	avatar: "https://fakeimages.com/images/fake.jpg"
}

const signedOutMocks = [
  {
    request: { query: CURRENT_USER_QUERY },
    result: { data: { me: null } },
  },
];

const signedInMocks = [
  {
    request: { query: CURRENT_USER_QUERY },
    result: {
			data: {
				me: {
					...mockUser
				}
			}
		}
  }
];

export { mockUser, signedInMocks, signedOutMocks };
