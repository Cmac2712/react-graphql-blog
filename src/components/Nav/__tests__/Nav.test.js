import React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { MockedProvider } from 'react-apollo/test-utils';
import wait from 'waait';
import { BrowserRouter } from 'react-router-dom';
import { CURRENT_USER_QUERY } from '../../User';
import Nav  from '../index';
import { mockUser, signedInMocks, signedOutMocks } from '../../../TestUtils';

describe('<Nav/>', () => {
	it('matches snapshot (logged out)', async () => {
		const wrapper = mount(
			<BrowserRouter>
				<MockedProvider mocks={signedOutMocks}>
					<Nav/>
				</MockedProvider>
			</BrowserRouter>
		);
		 await wait();
		 wrapper.update();
		
		expect(toJSON(wrapper.find('[data-test]'))).toMatchSnapshot();
	});

	it('matches snapshot (logged in)', async () => {
		const wrapper = mount(
			<BrowserRouter>
				<MockedProvider mocks={signedOutMocks}>
					<Nav/>
				</MockedProvider>
			</BrowserRouter>
		);
		 await wait();
		 wrapper.update();
		
		expect(toJSON(wrapper.find('[data-test]'))).toMatchSnapshot();
	});

	it('should say "Log In / Create an Account" when logged out', async () => {
		const wrapper = mount(
			<BrowserRouter>
				<MockedProvider mocks={signedOutMocks}>
					<Nav/>
				</MockedProvider>
			</BrowserRouter>
		);

		await wait();
		wrapper.update();
		const SignUpLink = wrapper.find('[href="/cms/create-post"]'); 

		expect(SignUpLink.text()).toBe("Log In / Create an Account");	
	});

	it('should say "Dashboard" when logged in', async () => {
		const wrapper = mount(
			<BrowserRouter>
				<MockedProvider mocks={signedInMocks}>
					<Nav/>
				</MockedProvider>
			</BrowserRouter>
		);

		await wait();
		wrapper.update();
		const SignUpLink = wrapper.find('[href="/cms/create-post"]'); 

		expect(SignUpLink.text()).toBe("Dashboard");	
	});
});
