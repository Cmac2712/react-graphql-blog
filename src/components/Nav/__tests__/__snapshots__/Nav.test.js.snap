// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav/> matches snapshot (logged in) 1`] = `
Array [
  <styled.nav
    data-test={true}
  >
    <StyledComponent
      data-test={true}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-ifAKCX",
            "isStatic": false,
            "lastClassName": "btIsrt",
            "rules": Array [
              "
	display: flex;
	align-items: center;
	//padding: ",
              [Function],
              "px;
	position: relative;
	z-index: 5;
	margin: 0 auto;
	max-width: ",
              [Function],
              ";
	width: 100%;
	padding: ",
              [Function],
              "px;

	.logo {
		font-family: ",
              [Function],
              ";
		font-size: 1.5rem;
		font-weight: bold;
		margin-right: auto;
		color: ",
              [Function],
              ";
		padding: 0;
	}
	
	a {
		color: ",
              [Function],
              ";
		padding: 0 ",
              [Function],
              "px;
	}
",
            ],
          },
          "displayName": "styled.nav",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-ifAKCX",
          "target": "nav",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <nav
        className="sc-ifAKCX btIsrt"
        data-test={true}
      >
        <Link
          className="logo"
          replace={false}
          to="/"
        >
          <a
            className="logo"
            href="/"
            onClick={[Function]}
          >
            React GraphQL Blog
          </a>
        </Link>
        <User>
          <Query
            query={
              Object {
                "definitions": Array [
                  Object {
                    "directives": Array [],
                    "kind": "OperationDefinition",
                    "name": undefined,
                    "operation": "query",
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "me",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "id",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "email",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "name",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "screenName",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "avatar",
                                },
                                "selectionSet": undefined,
                              },
                            ],
                          },
                        },
                      ],
                    },
                    "variableDefinitions": Array [],
                  },
                ],
                "kind": "Document",
                "loc": Object {
                  "end": 73,
                  "start": 0,
                },
              }
            }
          >
            <Link
              replace={false}
              to="/cms/create-post"
            >
              <a
                href="/cms/create-post"
                onClick={[Function]}
              >
                Log In / Create an Account
              </a>
            </Link>
          </Query>
        </User>
      </nav>
    </StyledComponent>
  </styled.nav>,
  <StyledComponent
    data-test={true}
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-ifAKCX",
          "isStatic": false,
          "lastClassName": "btIsrt",
          "rules": Array [
            "
	display: flex;
	align-items: center;
	//padding: ",
            [Function],
            "px;
	position: relative;
	z-index: 5;
	margin: 0 auto;
	max-width: ",
            [Function],
            ";
	width: 100%;
	padding: ",
            [Function],
            "px;

	.logo {
		font-family: ",
            [Function],
            ";
		font-size: 1.5rem;
		font-weight: bold;
		margin-right: auto;
		color: ",
            [Function],
            ";
		padding: 0;
	}
	
	a {
		color: ",
            [Function],
            ";
		padding: 0 ",
            [Function],
            "px;
	}
",
          ],
        },
        "displayName": "styled.nav",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-ifAKCX",
        "target": "nav",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <nav
      className="sc-ifAKCX btIsrt"
      data-test={true}
    >
      <Link
        className="logo"
        replace={false}
        to="/"
      >
        <a
          className="logo"
          href="/"
          onClick={[Function]}
        >
          React GraphQL Blog
        </a>
      </Link>
      <User>
        <Query
          query={
            Object {
              "definitions": Array [
                Object {
                  "directives": Array [],
                  "kind": "OperationDefinition",
                  "name": undefined,
                  "operation": "query",
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "me",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "id",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "email",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "name",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "screenName",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "avatar",
                              },
                              "selectionSet": undefined,
                            },
                          ],
                        },
                      },
                    ],
                  },
                  "variableDefinitions": Array [],
                },
              ],
              "kind": "Document",
              "loc": Object {
                "end": 73,
                "start": 0,
              },
            }
          }
        >
          <Link
            replace={false}
            to="/cms/create-post"
          >
            <a
              href="/cms/create-post"
              onClick={[Function]}
            >
              Log In / Create an Account
            </a>
          </Link>
        </Query>
      </User>
    </nav>
  </StyledComponent>,
  <nav
    className="sc-ifAKCX btIsrt"
    data-test={true}
  >
    <Link
      className="logo"
      replace={false}
      to="/"
    >
      <a
        className="logo"
        href="/"
        onClick={[Function]}
      >
        React GraphQL Blog
      </a>
    </Link>
    <User>
      <Query
        query={
          Object {
            "definitions": Array [
              Object {
                "directives": Array [],
                "kind": "OperationDefinition",
                "name": undefined,
                "operation": "query",
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "me",
                      },
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "id",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "email",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "name",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "screenName",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "avatar",
                            },
                            "selectionSet": undefined,
                          },
                        ],
                      },
                    },
                  ],
                },
                "variableDefinitions": Array [],
              },
            ],
            "kind": "Document",
            "loc": Object {
              "end": 73,
              "start": 0,
            },
          }
        }
      >
        <Link
          replace={false}
          to="/cms/create-post"
        >
          <a
            href="/cms/create-post"
            onClick={[Function]}
          >
            Log In / Create an Account
          </a>
        </Link>
      </Query>
    </User>
  </nav>,
]
`;

exports[`<Nav/> matches snapshot (logged out) 1`] = `
Array [
  <styled.nav
    data-test={true}
  >
    <StyledComponent
      data-test={true}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-ifAKCX",
            "isStatic": false,
            "lastClassName": "btIsrt",
            "rules": Array [
              "
	display: flex;
	align-items: center;
	//padding: ",
              [Function],
              "px;
	position: relative;
	z-index: 5;
	margin: 0 auto;
	max-width: ",
              [Function],
              ";
	width: 100%;
	padding: ",
              [Function],
              "px;

	.logo {
		font-family: ",
              [Function],
              ";
		font-size: 1.5rem;
		font-weight: bold;
		margin-right: auto;
		color: ",
              [Function],
              ";
		padding: 0;
	}
	
	a {
		color: ",
              [Function],
              ";
		padding: 0 ",
              [Function],
              "px;
	}
",
            ],
          },
          "displayName": "styled.nav",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-ifAKCX",
          "target": "nav",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <nav
        className="sc-ifAKCX btIsrt"
        data-test={true}
      >
        <Link
          className="logo"
          replace={false}
          to="/"
        >
          <a
            className="logo"
            href="/"
            onClick={[Function]}
          >
            React GraphQL Blog
          </a>
        </Link>
        <User>
          <Query
            query={
              Object {
                "definitions": Array [
                  Object {
                    "directives": Array [],
                    "kind": "OperationDefinition",
                    "name": undefined,
                    "operation": "query",
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "me",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "id",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "email",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "name",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "screenName",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "avatar",
                                },
                                "selectionSet": undefined,
                              },
                            ],
                          },
                        },
                      ],
                    },
                    "variableDefinitions": Array [],
                  },
                ],
                "kind": "Document",
                "loc": Object {
                  "end": 73,
                  "start": 0,
                },
              }
            }
          >
            <Link
              replace={false}
              to="/cms/create-post"
            >
              <a
                href="/cms/create-post"
                onClick={[Function]}
              >
                Log In / Create an Account
              </a>
            </Link>
          </Query>
        </User>
      </nav>
    </StyledComponent>
  </styled.nav>,
  <StyledComponent
    data-test={true}
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-ifAKCX",
          "isStatic": false,
          "lastClassName": "btIsrt",
          "rules": Array [
            "
	display: flex;
	align-items: center;
	//padding: ",
            [Function],
            "px;
	position: relative;
	z-index: 5;
	margin: 0 auto;
	max-width: ",
            [Function],
            ";
	width: 100%;
	padding: ",
            [Function],
            "px;

	.logo {
		font-family: ",
            [Function],
            ";
		font-size: 1.5rem;
		font-weight: bold;
		margin-right: auto;
		color: ",
            [Function],
            ";
		padding: 0;
	}
	
	a {
		color: ",
            [Function],
            ";
		padding: 0 ",
            [Function],
            "px;
	}
",
          ],
        },
        "displayName": "styled.nav",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-ifAKCX",
        "target": "nav",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <nav
      className="sc-ifAKCX btIsrt"
      data-test={true}
    >
      <Link
        className="logo"
        replace={false}
        to="/"
      >
        <a
          className="logo"
          href="/"
          onClick={[Function]}
        >
          React GraphQL Blog
        </a>
      </Link>
      <User>
        <Query
          query={
            Object {
              "definitions": Array [
                Object {
                  "directives": Array [],
                  "kind": "OperationDefinition",
                  "name": undefined,
                  "operation": "query",
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "me",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "id",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "email",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "name",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "screenName",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "avatar",
                              },
                              "selectionSet": undefined,
                            },
                          ],
                        },
                      },
                    ],
                  },
                  "variableDefinitions": Array [],
                },
              ],
              "kind": "Document",
              "loc": Object {
                "end": 73,
                "start": 0,
              },
            }
          }
        >
          <Link
            replace={false}
            to="/cms/create-post"
          >
            <a
              href="/cms/create-post"
              onClick={[Function]}
            >
              Log In / Create an Account
            </a>
          </Link>
        </Query>
      </User>
    </nav>
  </StyledComponent>,
  <nav
    className="sc-ifAKCX btIsrt"
    data-test={true}
  >
    <Link
      className="logo"
      replace={false}
      to="/"
    >
      <a
        className="logo"
        href="/"
        onClick={[Function]}
      >
        React GraphQL Blog
      </a>
    </Link>
    <User>
      <Query
        query={
          Object {
            "definitions": Array [
              Object {
                "directives": Array [],
                "kind": "OperationDefinition",
                "name": undefined,
                "operation": "query",
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "me",
                      },
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "id",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "email",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "name",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "screenName",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "avatar",
                            },
                            "selectionSet": undefined,
                          },
                        ],
                      },
                    },
                  ],
                },
                "variableDefinitions": Array [],
              },
            ],
            "kind": "Document",
            "loc": Object {
              "end": 73,
              "start": 0,
            },
          }
        }
      >
        <Link
          replace={false}
          to="/cms/create-post"
        >
          <a
            href="/cms/create-post"
            onClick={[Function]}
          >
            Log In / Create an Account
          </a>
        </Link>
      </Query>
    </User>
  </nav>,
]
`;
